aana/api/request_handler.py

```diff
@@ -95,16 +96,24 @@ async def is_ready(self):
         """
         return AanaJSONResponse(content={"ready": self.ready})
 
-    async def execute_task(self, task_id: str) -> Any:
+    async def check_health(self):
+        """Check the health of the application."""
+        # Heartbeat for the running tasks
+        with get_session() as session:
+            task_repo = TaskRepository(session)
+            task_repo.heartbeat(self.running_tasks)
```

Comment:
Is there any new PR for this? From the code, it just update the `update` field. And it does not throw any exceptions. It will works for the tasks but should we have something to check the health of request handler itself?

Reply:
No, that is all the code. The request handler updates `updated_at` field and task controller check the task health in `update_expired_tasks`. Request handler doesn't have any check because so far the main issue was that it was dying leaving the task hanging. For that you don't need any extra checks, Ray kills it and it stops sending out heartbeats.